components/elm/src/biogeophys/UrbanFluxesMod.F90

```diff
@@ -664,7 +715,36 @@ subroutine UrbanFluxes (bounds, num_nourbanl, filter_nourbanl,
             obu(l) = zldis(l)/zeta
          end do
 
-      end do   ! end iteration
+         ! Test for convergence
+         iter_final = iter
+         if (iter >= itmin) then
+            fnl_iter_old = fnl_iter
+            fnl_iter = 0
+            do fl = 1, fnl_iter_old
+               l = filterl_copy(fl)
+               if (.not. (abs(tau_diff(l)) < dtaumin)) then
+                  fnl_iter = fnl_iter + 1
+                  filterl_copy(fnl_iter) = l
+               end if
+            end do
+            if (fnl_iter == 0) then
+               exit ITERATION
+            end if
+            ! After weeding out landunits that have converged, we also need to
+            ! filter out the associated columns.
+            fnc_iter_old = fnc_iter
+            fnc_iter = 0
+            do fc = 1, fnc_iter_old
+               c = filterc_copy(fc)
+               l = col_pp%landunit(c)
+               if (.not. (abs(tau_diff(l)) < dtaumin)) then
+                  fnc_iter = fnc_iter + 1
+                  filterc_copy(fnc_iter) = c
+               end if
+            end do
+         end if
+
+      end do ITERATION ! end iteration
```

Comment:
Is this code active even when `implicit_stress=.false.`?

Reply:
Yes, but only on the last iteration, and it has no effect. I don't think there's any meaningful cost to doing it, but if you think it would be clearer, I could put an `if (implicit_stress)` block around it.

Reply:
Thanks for the explanation. There isn't a need to change things.